new_park features

nick-suitable opened the pull request:
added newpark.html, newpark.js, newpark.php
this allows form to be sumbitted and info to be stored in db... STILL
MISSING: attributes (facilities) and pictures.

Init.php also updated. the database is only missing a pictures table at
this commit.

Still need functions to return this info as JSON objects.

Commit:
new_park features

added newpark.html, newpark.js, newpark.php
this allows form to be sumbitted and info to be stored in db... STILL
MISSING: attributes and pictures.

Init.php also updated. the database is only missing a pictures table at
this commit.

Still need functions to return this info

nick-suitable commented:
How will the new park handle the input of attributes? Basically, I'm asking how the users are inputting that the park has swings, jungle gyms, tracks, etc. Will there be a set number of textboxes? dynamically add text boxes so they can input as many as they want? That's why I left that alone for now... There is a table in the db for attributes though. 

Also on the to-do list for back-end:
- Getting an average of the reviews to display on the park page
- creating functions to return this info from db to client

sag131 commented on pgh_play/www/php/interface.php:
```diff
@@ -201,5 +203,72 @@ function get_parks(){
+// this will store the following info in the db: Park name, address, rating, ages, comments
+//  all of these inputs are received by the new park form a user fills out
+// *****facilities and pictures still need to be implemented 
+function new_park(){
```
Instead of taking a JSON object and decoding it into a php object   (e.g.  json_decode($_POST['obj'])), this seems less structured.   Also, it looks like this one function is doing multiple things, like taking a rating, setting fixed attributes, etc.

nick-suitable replied:
I created the function to coincide with the simple new park form I whipped together. I agree that it probably needs refactored, but I wanted to get the functionality working with the db before cleaning it up and drilling down more into separate functions

sag131 replied:
Something like this might do the trick:
{ "name":"park name",
 "address":"street addr",
 "city":"city name"
 "gps_long":234.234234,
 "gps_lat":234.234234,
 "rating":4.2,
 "infant_safe":true,
 "toddler_safe":true,
 "five_eight":true,
 "nine_twelve":true,
 attributes: [  1, 2, 3, 4, 5, 6, 7 ]
}

In javascript,  var output;  output['name'] = "Green Tree Park";  ...  output['attributes'] = new Array();  output['attributes'][0] = 1; ...  (arrays need to start with 0 for proper json encoding)

In php,  $obj = json_decode($_POST['obj'], true);
$obj will have same structure as the var output did.

sag131 commented on pgh_play/www/php/interface.php:
```diff
@@ -201,5 +203,72 @@ function get_parks(){
+		$result = $db->query($query) or trigger_error(mysql_error(). " ".$query);
+	}
+	// store info about appropriate age groups for park if set
+	if (strcmp($_POST['infants'],"yes") == 0){
```
These seem like fixed attributed as opposed to the dynamic ones she spoke of.

nick-suitable replied:
I was setting the age attributes under the impression that these particular ones were fixed. Basically not allowing to set their own age ranges, but giving a fair estimate on which ages they deemed appropriate via checkbox. The other attributes would be dynamic, such as swings and tracks etc.

sag131 commented on pgh_play/www/php/interface.php:
```diff
@@ -201,5 +203,72 @@ function get_parks(){
+		echo "Thanks!";
+	}
+	else{
+		echo false;
```
Change to die(false); ?

Because if info not set, still doing the code below.

nick-suitable replied:
Makes sense, will update.

nick-suitable commented:
Samit, if you have time to start putting together functions to return information in the new tables created, that would be a big help for me. They would resemble the ones created for get_neighborhood and get_parks I would imagine, just returning the JSON objects. If not, I can start on them a little later or in the AM. 

Also, let me know what else we can add to the back-end to get it wrapped up as soon as possible.

sag131 committed:
Refactor new_park().  Create get_park_info()

sag131 committed:
Updated DB Schema, added commenting functions, added rating to get_park_info()

Fixed a small error in the get_park_info

sag131 committed:
Added function to get list of attributes

sag131 committed:
Added .php pages for calling these functions from client

sag131 committed:
updated json object

Commit:
minor fixes to remove errors

Looks good I think. Can't run it fully because now the newpark.js isn't
integrated correctly. The new functions need an obj parameter now.
